docs/notes.md

```diff
@@ -603,4 +603,254 @@ sends a pair \\((L\_j,R\_j)\\) of points at each step \\(j = k\dots1\\). An
+  \mathbf{y}^n\_{(j)} &= \mathbf{y}^{n \cdot m}\_{[j \cdot n : (j+1) \cdot n - 1]} \\\\
+  z_{(j)} &= z^j
+\end{aligned}
+\\]
+
+Therefore, we can simplify the following statements:
+
+\\[
+\begin{aligned}
+  \sum_{j=0}^{m-1}{\langle {\mathbf{l}}\_{(j)}(x), {\mathbf{G}\_{(j)}} \rangle} &= {\langle {\mathbf{l}}\_{(0)}(x), {\mathbf{G}}\_{(0)} \rangle} + 
+    {\langle {\mathbf{l}}\_{(1)}(x), {\mathbf{G}}\_{(1)} \rangle} + 
+    \dots + 
+    {\langle {\mathbf{l}}\_{(m-1)}(x), {\mathbf{G}}\_{(m-1)} \rangle}\\\\
+  &= {\langle {\mathbf{l}}\_{(0)}(x) || {\mathbf{l}}\_{(1)}(x) || \dots || {\mathbf{l}}\_{(m-1)}(x), {\mathbf{G}}\_{(0)} || {\mathbf{G}}\_{(1)} || \dots || {\mathbf{G}}\_{(m-1)} \rangle} \\\\
+  &= {\langle {\mathbf{l}}(x), {\mathbf{G}} \rangle} \\\\
+  \sum_{j=0}^{m-1}{\langle {\mathbf{r}}\_{(j)}(x) \circ (\mathbf{y}^{n}\_{(j)})^{-1}, {\mathbf{H}}\_{(j)} \rangle} &= {\langle {\mathbf{r}}\_{(0)}(x) \circ (\mathbf{y}^{n}\_{(0)})^{-1}, {\mathbf{H}}\_{(0)} \rangle} + 
+    {\langle {\mathbf{r}}\_{(1)}(x) \circ (\mathbf{y}^{n}\_{(1)})^{-1}, {\mathbf{H}}\_{(1)} \rangle} + 
+    \dots + 
+    {\langle {\mathbf{r}}\_{(m-1)}(x) \circ (\mathbf{y}^{n}\_{(m-1)})^{-1}, {\mathbf{H}}\_{(m-1)} \rangle} \\\\
+  &= {\langle {\mathbf{r}}\_{(0)}(x) \circ (\mathbf{y}^{n}\_{(0)})^{-1} || {\mathbf{r}}\_{(1)}(x) \circ (\mathbf{y}^{n}\_{(1)})^{-1} || \dots || {\mathbf{r}}\_{(m-1)}(x) \circ (\mathbf{y}^{n}\_{(m-1)})^{-1}, {\mathbf{H}}\_{(0)} || {\mathbf{H}}\_{(1)} || \dots || {\mathbf{H}}\_{(m-1)}  \rangle}\\\\
+  &= {\langle {\mathbf{r}}\_{(0)}(x) \circ (\mathbf{y}^{n \cdot m}\_{[0 : n - 1]})^{-1} || {\mathbf{r}}\_{(1)}(x) \circ (\mathbf{y}^{n \cdot m}\_{[ n : 2 \cdot n - 1]})^{-1} || \dots || {\mathbf{r}}\_{(m-1)}(x) \circ (\mathbf{y}^{n \cdot m}\_{[(m-1) \cdot n : m \cdot n - 1]})^{-1}, {\mathbf{H}}\_{(0)} || {\mathbf{H}}\_{(1)} || \dots || {\mathbf{H}}\_{(m-1)}  \rangle}\\\\
+  &= {\langle  {\mathbf{r}}(x) \circ {\mathbf{y}^{-n \cdot m}}, {\mathbf{H}} \rangle}
+\end{aligned}
+\\]
+
+We can combine the values and commitments from all the `m` parties by summing them directly:
+
+\\[
+\begin{aligned}
+  {\widetilde{e}} &= \sum_{j=0}^{m-1} {\widetilde{e}\_{(j)}} \\\\
+  A &= \sum_{j=0}^{m-1} A_{(j)} \\\\
+  S &= \sum_{j=0}^{m-1} S_{(j)} \\\\
+\end{aligned}
+\\]
+
+With these observations, we can simplify the combined `m`-party statement about \\({\mathbf{l}}(x)\\) and \\({\mathbf{r}}(x)\\) into:
+
+\\[
+\begin{aligned}
+  {\langle {\mathbf{l}}(x), {\mathbf{G}} \rangle} + {\langle {\mathbf{r}}(x) \circ {\mathbf{y}^{-n \cdot m}} , {\mathbf{H}} \rangle} \stackrel{?}{=} -{\widetilde{e}} {\widetilde{B}} + A + x S - z{\langle {\mathbf{1}}, {\mathbf{G}} \rangle} + z{\langle {\mathbf{1}}, {\mathbf{H}} \rangle} + \sum_{j=0}^{m-1} {\langle z^{j+2} \cdot (\mathbf{y}^{n \cdot m}\_{[j \cdot n : (j+1) \cdot n - 1]})^{-1} \circ {\mathbf{2}}^n, {\mathbf{H}}\_{[j \cdot n : (j+1) \cdot n - 1]} \rangle} 
```

Comment:
The H' terms
```
(\mathbf{y}^{n \cdot m}\_{[j \cdot n : (j+1) \cdot n - 1]})^{-1} \circ {\mathbf{2}}^n
```
are pretty inscrutable (at least to me), but the way you described it in the code seems more clear, maybe it's possible to factor out this part into a new vector that's built up in the same way as in the code?

Reply:
I'm confused where you mean by "in the code" - in the inner-product-protocol documentation? Or in the comments in the code? (I don't think I wrote any extensive comments explaining the math in the code)

Either way, I'm guessing what you mean is to use the `H'` vector instead of expanding that to `H` and the `y` exponent - I'll give that a shot!

Reply:
It looks much clearer, I like it! 
👍